### ISSUE TITLE
SendMessageback is interrupting speech dialog

### ISSUE THREAD

**Issue by ZaherAirout** [microsoft/BotFramework-WebChat#2616]
## Screenshots
![image](https://user-images.githubusercontent.com/11921609/69221370-c6115c00-0b7f-11ea-8b8b-7156a57c7c18.png)

## Version
CDN latest package.

## Describe the bug
I need to send a specific data to the bot behind the scenes. to tell the bot which options the user can selects, I tried SEND_EVENT but its sending in another dialog not the current one, So I'm using SEND_MESSAGE_BACK to send these options to the bot from my javascript code like this
 ```javascript
dispatch({
    type: 'WEB_CHAT/SEND_MESSAGE_BACK',
    payload: {
        text: result,
        method: "speech",
    }
});
```
the bug appears when the user is interacting using speech, when I send the MESSAGE_BACK the bot stop speaking and the dialog waits for user keyboard input not for speech.

## To Reproduce

1. Start a speech conversation with a bot
2. Send a MESSAGE_BACK activity
3. the bot stop speaking and stop listening to the user after that

## Expected behavior
bot receives the MESSAGE_BACK and continue speakinig and listening to the user.

## Additional context
I cloned this project and I found something maybe related to this issue.
in the [sendMessageBack.js](https://github.com/microsoft/BotFramework-WebChat/blob/39ab54efe9c15df7308300e9474d946acf552b4b/packages/core/src/actions/sendMessageBack.js#L6) you didn't send the method key and it becomes 'keyboard' by default in [postActivity.js](https://github.com/microsoft/BotFramework-WebChat/blob/39ab54efe9c15df7308300e9474d946acf552b4b/packages/core/src/actions/postActivity.js#L6)

[Bug]

**Comment by corinagum** [comment]
Assigning to @CoHealer in lieu of @stevkan.

**Comment by stevkan** [comment]
The option you want to use is `DIRECT_LINE/POST_ACTIVITY` which allows you to alter the activity from the user before it is sent to the bot. Once you have the below configured to how you want, then just pass `store` as a property to `window.ReactDOM.render()` (or `window.WebChat.renderWebChat()`). When it is received by the bot, you can pass the data into a new activity/prompt.

**index.html:**
```js
const store = window.WebChat.createStore( {}, ( { dispatch } ) => next => async action => {
  if ( action.type === 'DIRECT_LINE/POST_ACTIVITY' ) {
    let text = action.payload.activity.text.toLowerCase();
    if ( text === 'sending details.' ) {
      const userId = 'xyz789';
      const userDetails = { 'name': 'Bob', 'age': 35, 'features': { 'eyes': 'brown', 'hair': 'blonde' } };
      action = window.simpleUpdateIn(
        action,
        [ 'payload', 'activity', 'channelData' ],
        () => ( {
          'userId': userId,
          'userDetails': userDetails
        } )
      )
    }
  }
  return next( action );
});
```

**mainBot.js:**
```js
this.onMessage(async (context, next) => {
  const { activity } = context;

  if (activity.channelData && activity.channelData.userId) {
    const userDetails = activity.channelData.userDetails;
    await context.sendActivity(`It looks like you are ${ userDetails.age } old, ${ userDetails.name }. You have ${ userDetails.features.eyes } eyes and ${ userDetails.features.hair }.`);
  }
  await next();
});
```

Hope of help!

![image](https://user-images.githubusercontent.com/18403946/69389665-8a5ebb00-0c81-11ea-8475-457f06623fa6.png)

![image](https://user-images.githubusercontent.com/18403946/69389719-b8dc9600-0c81-11ea-9e68-c9720b1ce540.png)

**Comment by stevkan** [comment]
As for `WEB_CHAT/SEND_MESSAGE_BACK`, I will attempt to repro.

**Comment by ZaherAirout** [comment]
> The option you want to use is `DIRECT_LINE/POST_ACTIVITY` which allows you to alter the activity from the user before it is sent to the bot. Once you have the below configured to how you want, then just pass `store` as a property to `window.ReactDOM.render()` (or `window.WebChat.renderWebChat()`). When it is received by the bot, you can pass the data into a new activity/prompt.
> 
> **index.html:**
> 
> ```js
> const store = window.WebChat.createStore( {}, ( { dispatch } ) => next => async action => {
>   if ( action.type === 'DIRECT_LINE/POST_ACTIVITY' ) {
>     let text = action.payload.activity.text.toLowerCase();
>     if ( text === 'sending details.' ) {
>       const userId = 'xyz789';
>       const userDetails = { 'name': 'Bob', 'age': 35, 'features': { 'eyes': 'brown', 'hair': 'blonde' } };
>       action = window.simpleUpdateIn(
>         action,
>         [ 'payload', 'activity', 'channelData' ],
>         () => ( {
>           'userId': userId,
>           'userDetails': userDetails
>         } )
>       )
>     }
>   }
>   return next( action );
> });
> ```
> 
> **mainBot.js:**
> 
> ```js
> this.onMessage(async (context, next) => {
>   const { activity } = context;
> 
>   if (activity.channelData && activity.channelData.userId) {
>     const userDetails = activity.channelData.userDetails;
>     await context.sendActivity(`It looks like you are ${ userDetails.age } old, ${ userDetails.name }. You have ${ userDetails.features.eyes } eyes and ${ userDetails.features.hair }.`);
>   }
>   await next();
> });
> ```
> 
> ![image](https://user-images.githubusercontent.com/18403946/69389665-8a5ebb00-0c81-11ea-8475-457f06623fa6.png)
> 
> ![image](https://user-images.githubusercontent.com/18403946/69389719-b8dc9600-0c81-11ea-9e68-c9720b1ce540.png)
> 
> Hope of help!

thank you @stevkan. that's working, now I'm sending a new `DIRECT_LINE/POST_ACTIVITY` with no text to hide the message from the web chat. but the problem still exist in `WEB_CHAT/SEND_MESSAGE_BACK`
```javascript
dispatch({
    type: "DIRECT_LINE/POST_ACTIVITY",
    meta: {
        method: lastMethod || "speech"
    },
    payload: {
        activity: {
            text: "",
            textFormat: "plain",
            type: "message",
            channelData: {
                result: result
            }
        }
    }
});
```

**Comment by stevkan** [comment]
@ZaherAirout, can you post an example of the `result` object that you are passing to the `text` property?

Also, just verifying that you were using the v4.6.0 of Web Chat?

**Comment by ZaherAirout** [comment]
I'm using the [latest version of web chat](https://cdn.botframework.com/botframework-webchat/latest/webchat.js).
I'm sending the result as a JSON string.
like this code:
```javascript
let result = JSON.stringify({
    requestId: requestId + '_FULFILLED',
    data: [
        {
            value: 'value',
            title: 'title'
        },
    ]
});
dispatch({
    type: 'WEB_CHAT/SEND_MESSAGE_BACK',
    payload: {
        text: result,
    }
});
```
![image](https://user-images.githubusercontent.com/11921609/69706462-6fb49800-1100-11ea-82b3-63e84732e479.png)

**Comment by stevkan** [comment]
I'm not able to repro the issue. When I click the microphone and speak, I can do so uninterrupted (I even tested by speaking a long paragraph).

I also tested by speaking once followed immediately by another speech input. Both inputs registered along with the `SEND_MESSAGE_BACK` payload for each. Again, speech was not interrupted by waiting for keyboard input.

One thing I noticed is the number and order of pertinent actions is different from mine to yours. I don't know if the differing order is a result of where / how we are logging, but in my case the actions happen together: two send message actions, two post actions, two post pending actions, and later (not captured in the image) two post fulfilled actions. Additionally, the `method: "keyboard"` shouldn't be a factor as that property is merely a flag indicating how the input was captured.

![image](https://user-images.githubusercontent.com/18403946/69772498-f3907380-1144-11ea-98f5-95ab863f405e.png)

Regarding your other "in between" actions, are there any other methods being called, properties being set, or values being passed that might be affecting this? Are you able to post your web chat code so I can better diagnose?
